client/network/src/protocol.rs

```diff
@@ -78,7 +78,9 @@ const PROPAGATE_TIMEOUT: time::Duration = time::Duration::from_millis(2900);
 /// Maximim number of known block hashes to keep for a peer.
 const MAX_KNOWN_BLOCKS: usize = 1024; // ~32kb per peer + LruHashSet overhead
 /// Maximim number of known extrinsic hashes to keep for a peer.
-const MAX_KNOWN_EXTRINSICS: usize = 4096; // ~128kb per peer + overhead
+///
+/// This should be approx. 2 blocks full of transactions for the network to function properly.
+const MAX_KNOWN_EXTRINSICS: usize = 10240; // ~300kb per peer + overhead.
```

Comment:
how about chains with longer block time / larger tx per block?

Reply:
@xlc let's make an issue for that

This should not be a const but rather some sort of configuration (runtime/cli or deduced from other values)

Reply:
>  larger tx per block

It only stores the hash of a transaction.

Reply:
I mean larger number of tx per block